demos/python_demos/whiteboard_inpainting_demo/utils/ie_tools.py

```diff
@@ -0,0 +1,92 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import cv2
+import logging as log
+import numpy as np
+import os
+import sys
+
+from openvino.inference_engine import IENetwork, IECore # pylint: disable=import-error,E0611
+
+
+class IEModel:
+    """Class for inference of models in the Inference Engine format"""
+    def __init__(self, ie, model_path, conf=.6, device='CPU', ext_path='', num_reqs=1):
+        self.confidence = conf
+        self.load_ie_model(ie, model_path, device, None, ext_path, num_reqs)
+
+    def _preprocess(self, img):
+        _, _, h, w = self.get_input_shape().shape
+        img = np.expand_dims(cv2.resize(img, (w, h)).transpose(2, 0, 1), axis=0)
+        return img
+
+    def forward(self, img):
+        """Performs forward pass of the wrapped IE model"""
+        res = self.net.infer(inputs={self.input_key: self._preprocess(img)})
+        return np.copy(res[self.output_key])
+
+    def get_detections(self, input):
+        raise NotImplementedError
+
+    def get_input_shape(self):
+        """Returns an input shape of the wrapped IE model"""
+        return self.inputs_info[self.input_key]
+
+    def get_allowed_inputs_len(self):
+        return (1, )
+
+    def get_allowed_outputs_len(self):
+        return (1, )
+
+    def load_ie_model(self, ie, model_xml, device, plugin_dir, cpu_extension='', num_reqs=1):
+        """Loads a model in the Inference Engine format"""
+        model_bin = os.path.splitext(model_xml)[0] + ".bin"
+        # Plugin initialization for specified device and load extensions library if specified
+        log.info("Initializing Inference Engine plugin for %s ", device)
+
+        if cpu_extension and 'CPU' in device:
+            ie.add_extension(cpu_extension, 'CPU')
+        # Read IR
+        log.info("Loading network files:\n\t%s\n\t%s", model_xml, model_bin)
+        net = IENetwork(model=model_xml, weights=model_bin)
```

Comment:
Reading network using constructor is deprecated. Please, use `IECore.read_network()` method instead

Reply:
Fixed.